.github/workflows/tests.yml

```diff
@@ -262,6 +259,10 @@ jobs:
 
   macos-14:
     name: macOS 14.0 (Sonoma)
+    # TODO: when actions supports using ${{env}} in job.*.if and not just job.*.steps.*.if, use this:
+    #if: ${{ env.NIGHTLY }}
+    # in the meantime, copy-paste this:
+    if: ${{ github.event_name == 'schedule' || github.event_name == 'push' || github.event_name == 'workflow_dispatch' }}
```

Comment:
Just to double-check my understanding: the commit description for this change says that it's **disabling** the macos-14 runner, but these lines actually **enable** the macos-14 runner, right? Specifically, it becomes enabled for pull_request events?

I think it's still true that merging this PR would reduce the number of macos runners for pull_request events from 3 to 2.

Reply:
Ah! Sorry for the confusion... 

Without this line, the macos-14 runner would run unconditionally. Whereas, with this line, the macos-14 runner only runs for the triggers listed. And, since PRs aren't listed, it gets disabled for PRs.

Reply:
It's almost like a double-negative? It might be a little more clear if it was instead worded like this:

```
if ${{ github.event_name != 'pull_request' }}
```

That way it's clearer that it runs on every trigger _but_ PRs.

Reply:
Oh wow, I read the diff backwards, sorry! For some reason I was convinced that these lines got removed instead of added :facepalm: